### DISCUSSION TITLE
RFC - Rename master branch to main

### DISCUSSION THREAD

**Discussion by xclaesse** [mesonbuild/meson#11178] [Ideas]
Many open source project decided to rename their "master" branch to "main" over the past years. For consistency with the rest of the community, it might be a good idea to do the same in the Meson project. Are there objections?

Likewise, from a quick grep, we use the term "master" project (in the context of sub-projects) in a few places in our documentation, I think we should rather use "main" project.

Note: I don't want to enter any political/ethical discussion, I personally see this mostly as consistency with the rest of the community. I might be biased, but I see most of GNOME world, GStreamer and Mesa, did that change at least. Those are probably the biggest Meson users.

**Comment by dcbaker** [comment]
Freedesktop has mostly moved away from "master" at this point, mostly to "main" (though I think there's a few projects with the "develop" and "release" branches).

I'm in favor of the change, given that "main" is:
1) now the default for new projects on github
2) seems to be agreed upon as the new default
3) makes more sense than "master" (I was seriously confused by why "master" was the name of the primary branch for *years*)
4) I already have my local branch tracking upstream called "main" because muscle memory

I am, however *not* in favor of a change in name that is not "main", I'd rather keep "master" than change to something even more confusing like "trunk".

**Comment by eli-schwartz** [comment]
I don't see how this really adds any consistency. It causes havoc with existing clones.

And I actually despise the name with a godawful passion, but not for ideological reasons! It simply fails at achieving the stated goal. The name isn't semantically meaningful, but it does break tab completion with projects that have a "maint" branch. My main branch is the branch that I mainly use, which is different from the one that tracks the upstream golden state -- and the proper name to use is of course "trunk", which also semantically means the same thing that "master" semantically means. But that would be inconsistent. ;)

**Comment by eli-schwartz** [comment]
Just want to reiterate in case it gets lost in my other comment:

I am strongly opposed to any project in the world changing the name of the default branch in an existing repository no matter what it used to be...

***Until*** git actually learns to handle this sanely and not break existing clones. Until then, I'm sorry. But I think it's the sign of an insane mind to change the branch name.

**Reply by jpakkane** [reply]
Expecting Git to do _anything_ that would improve usability is not a winning proposition, sadly.

**Comment by tp-m** [comment]
I think Meson should change to `main` too. I'm surprised it's still on `master` every time.

The 1.0 release seems like a good opportunity to make the switch to me (although it could be at any point of course).

That way you leave a stale `master` behind for a while on a defined version, and then you can remove it at your leisure.

Heaps of projects have gone through this. It's not a big deal in practice, even if it would be nice if github/gitlab supported this better. 

Developers need a day to fix their muscle memory and then it's fine. New people get the new branch with checkouts.

**Reply by eli-schwartz** [reply]
Leaving a stale `master` behind for a while is the ***worst*** option of all.

**Comment by jpakkane** [comment]
The default branch name should optimally be the same everybody else uses. As that is not possilbe the next best choice is the same name that is used by most other projects. If that is "main", then so be it.

**Reply by eli-schwartz** [reply]
Why does it "have" to be the same everyone else uses? There's no shortage of projects using either one. In fact, there's probably more people using "master" than "main" :p because there's decades more repositories created that way.

I continue to maintain the optimal thing is to not break existing clones, which is something that actually affects users negatively -- and disproportionately affects less experienced users.

**Comment by nirbheek** [comment]
Since everyone else is commenting, I thought I'd chip in too.

I have no preference either way on this. My annoyance level will slightly increase in the short term and reduce in the long term if the trend of moving to main continues.

I think it's a goddamn shame that there has been no movement anywhere on adding support to git for branch name aliasing or changing of the default branch for a repository, but our entire profession exists to suffer due to bad tooling, so be it.
